src/Owin.Security.Providers.CanvasLMS/CanvasAuthenticationHandler.cs

```diff
@@ -228,5 +231,83 @@ protected override Task ApplyResponseChallengeAsync()
 
             return context.IsRequestCompleted;
         }
+
+        private async Task<object> RequestToken(string grantType, string grantToken, string grantTokenParameterName = null)
+        {
+            var requestPrefix = Request.Scheme + "://" + Request.Host;
```

Comment:
Avoiding string interp here is a micro-optimization, since this will be compiled as a single `String.Concat(a,b,c)`. :+1: for `Uri.SchemeDelimiter`, had not seen that before.

FWIW, this is all copy-pasta from another provider: https://github.com/NewBoCo/OwinOAuthProviders/commit/97bd415cf3ca2b82b753a06fa9897bb9bb00d6a1#diff-fa49c7867850d1aebd1ca41d9af79d36R65